scripts/us_census/pep_srh/process.py

```diff
@@ -0,0 +1,1057 @@
+    df1['HBA'] = (df['HBA_MALE'] + df['HBA_FEMALE']).copy()
+    df1['HIA'] = (df['HIA_MALE'] + df['HIA_FEMALE']).copy()
+    df1['HAA'] = (df['HAA_MALE'] + df['HAA_FEMALE']).copy()
+    df1['HNA'] = (df['HNA_MALE'] + df['HNA_FEMALE']).copy()
+    df1['HTOM'] = (df['HTOM_MALE'] + df['HTOM_FEMALE']).copy()
+    df1.to_csv(output_file_path + output_file_name, header=True, index=False)
+
+
+def process_national_files_2010_2020():
+    '''
+    Process national files from 2010 - 2020
+    Uses state level file as input
+    '''
+    # Section 1 - Writing As Is Data
+    input_file_path = _CODEDIR + PROCESS_AS_IS_DIR + '2010_2020/state/'
+    output_file_path = _CODEDIR + PROCESS_AS_IS_DIR + '2010_2020/national/'
+    output_file_name = 'national_2010_2020.csv'
+    files_list = os.listdir(input_file_path)
+    files_list.sort()
+
+    df = pd.read_csv(input_file_path + files_list[0])
+    df['LOCATION'] = 'country/USA'
+    df = df.groupby(['YEAR', 'LOCATION']).agg('sum').reset_index()
+    df.to_csv(output_file_path + output_file_name, index=False)
+
+    # Section 2 - Writing Agg data
+    input_file_path = _CODEDIR + PROCESS_AGG_DIR + '2010_2020/state/'
+    output_file_path = _CODEDIR + PROCESS_AGG_DIR + '2010_2020/national/'
+    files_list = os.listdir(input_file_path)
+    files_list.sort()
+
+    df = pd.read_csv(input_file_path + files_list[0])
+    df['LOCATION'] = 'country/USA'
+    df = df.groupby(['YEAR', 'LOCATION']).agg('sum').reset_index()
+    df.to_csv(output_file_path + output_file_name, index=False)
+
+
+def process_national_files():
+    '''
+    Process national files from 1980 - 2020
+    '''
+    process_national_files_1980_1990()
+    process_national_files_1990_2000()
+    process_national_files_2000_2010()
+    process_national_files_2010_2020()
+
+
+def process_state_files_1980_1990(download_dir):
+    '''
+    Process state files from 1980 - 1990
+    Age is present in bracket for 5 years, these are added up
+    '''
+    # Section 1 - Writing As Is data
+    input_file_path = _CODEDIR + download_dir + '1980_1990/state/'
+    output_file_path = _CODEDIR + PROCESS_AS_IS_DIR + '1980_1990/state/'
+    output_file_name = 'state_1980_1990.csv'
+    output_temp_file_name = 'state_1980_1990_temp.csv'
+    files_list = os.listdir(input_file_path)
+    files_list.sort()
+
+    column_names = [
+        "LOCATION", "YEAR_TEMP", "RACE_ORIGIN", "SEX", "A01", "A02", "A03",
+        "A04", "A05", "A06", "A07", "A08", "A09", "A10", "A11", "A12", "A13",
+        "A14", "A15", "A16", "A17", "A18"
+    ]
+
+    column_specification = [(0, 2), (2, 3), (3, 4), (4, 5), (5, 12), (12, 19),
+                            (19, 26), (26, 33), (33, 40), (40, 47), (47, 54),
+                            (54, 61), (61, 68), (68, 75), (75, 82), (82, 89),
+                            (89, 96), (96, 103), (103, 110), (110, 117),
+                            (117, 124), (124, 131)]
+
+    df = pd.read_fwf(input_file_path + files_list[0],
```

Comment:
Some files had spurious characters in other imports.
Does the read_fwf() handle such characters well? 
Are the files guaranteed to not be shifted when there are spurious characters?

Reply:
Input files which I have considered are in good shape. I have manually verified data data by aggregating and comparing them against Pandas output, they are matching as well.